HousingSearchApi.Tests/V2/E2ETests/AssetSearchTests.cs

```diff
@@ -47,71 +77,80 @@ public async Task ReturnsNoResultsWhenNoMatches()
         root.GetProperty("total").GetInt32().Should().Be(0);
     }
 
-    [Theory]
-    [InlineData("Gge 7 Toby Crossroad", "71312edd-10c3-41cf-b298-b97cce0c0123")]
-    [InlineData("6 Philip Alley", "5a6067de-458c-40cb-93a3-d16fe39da12a")]
-    [InlineData("Room 3 2 Kevin Inlet", "b5e85565-6241-4dd2-8f1f-50d626b129f2")]
-    public async Task ReturnsRelevantResultFirstByAddress(string searchText, string expectedReturnedId)
+    [Fact]
+    public async Task ReturnsRelevantResultFirstByAddress()
     {
-        // Arrange
-        var request = CreateSearchRequest(searchText);
-
-        // Act
-        var response = await _httpClient.SendAsync(request);
-
-        // Assert
-        response.StatusCode.Should().Be(HttpStatusCode.OK);
-        var root = GetResponseRootElement(response);
-        root.GetProperty("total").GetInt32().Should().BeGreaterThan(0);
-        var firstResult = root.GetProperty("results").GetProperty("assets")[0];
-        firstResult.GetProperty("id").GetString().Should().Be(expectedReturnedId);
+        foreach (var _ in Enumerable.Range(0, 10))
+        {
+            // Arrange
+            var randomAsset = RandomAsset();
+            var expectedReturnedId = randomAsset.GetProperty("id").GetString();
+            var request = CreateSearchRequest(randomAsset.GetProperty("assetAddress").GetProperty("addressLine1").GetString());
+
+            // Act
+            var response = await _httpClient.SendAsync(request);
+
+            // Assert
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+            var root = GetResponseRootElement(response);
+            root.GetProperty("total").GetInt32().Should().BeGreaterThan(0);
+            var firstResult = root.GetProperty("results").GetProperty("assets")[0];
+            firstResult.GetProperty("id").GetString().Should().Be(expectedReturnedId);
+        }
     }
 
-    [Theory]
-    [InlineData("NW1 3LF")]
-    [InlineData("L22 9BQ")]
-    [InlineData("PA3R 7JR")]
-    public async Task ReturnsRelevantResultFirstByPostcode(string searchTextPostcode)
+    [Fact]
+    public async Task ReturnsRelevantResultFirstByPostcode()
     {
-        // Arrange
-        var request = CreateSearchRequest(searchTextPostcode);
-
-        // Act
-        var response = await _httpClient.SendAsync(request);
-
-        // Assert
-        response.StatusCode.Should().Be(HttpStatusCode.OK);
-        var root = GetResponseRootElement(response);
-        root.GetProperty("total").GetInt32().Should().BeGreaterThan(0);
-        var firstResult = root.GetProperty("results").GetProperty("assets")[0];
-        firstResult.GetProperty("assetAddress").GetProperty("postCode").GetString().Should().Be(searchTextPostcode);
+        foreach (var _ in Enumerable.Range(0, 10))
+        {
+            // Arrange
+            var randomAsset = RandomAsset();
```

Comment:
Wasn't obvious what was happening here until after I spent a couple of minutes on it.

You can still do this, but cleaner:
```cs
var randomAssetPostCodes = GetListOfRandomAssetPostcodesFromFixture();

foreach (var postcode in randomAssetPostCodes)
{
    // search by postcode
    // parse result
    // check result
}
```

Would say this makes it clearer what's happening here.